src/Microsoft.Extensions.FileProviders.Physical/PhysicalFilesWatcher.cs

```diff
@@ -137,54 +190,47 @@ private void ReportChangeForMatchedEntries(string pattern)
             }
         }
 
-        private string NormalizeFilter(string filter)
+        private static string NormalizePath(string filter) => filter.Replace(Path.DirectorySeparatorChar, '/');
+
+        private static bool IsDirectoryPath(string path)
         {
-            // If the searchPath ends with \ or /, we treat searchPath as a directory,
-            // and will include everything under it, recursively.
-            if (IsDirectoryPath(filter))
+            return path.Length > 0
```

Comment:
Added a null check to https://github.com/aspnet/FileSystem/pull/183/files#diff-8bb8e51541651d6e13df23d9a7311bb1R44. The calling code also perfoms a null check so it really isn't a valid input here.